bitcoin/src/address/mod.rs

```diff
@@ -273,6 +273,16 @@ impl From<Network> for KnownHrp {
     fn from(n: Network) -> Self { Self::from_network(n) }
 }
 
+impl From<KnownHrp> for NetworkKind {
+    fn from(hrp: KnownHrp) -> Self {
+        match hrp {
+            KnownHrp::Mainnet => NetworkKind::Main,
+            KnownHrp::Testnets => NetworkKind::Test,
+            KnownHrp::Regtest => NetworkKind::Test,
+        }
+    }
+}
```

Comment:
I feel somewhat uneasy about this conversion since it's lossy. But not really wrong.

Reply:
`KnownHrp` is a larger type than `NetworkKind` so it's in the type signature that it'll be lossy.